mmcls/models/losses/cross_entropy_loss.py

```diff
@@ -18,15 +18,63 @@ def cross_entropy(pred, label, weight=None, reduction='mean', avg_factor=None):
     return loss
 
 
+def binary_cross_entropy(pred,
+                         label,
+                         weight=None,
+                         reduction='mean',
+                         avg_factor=None):
+    """Calculate the binary CrossEntropy loss with logits.
+
+    Args:
+        pred (torch.Tensor): The prediction with shape (N, C), where C is the
+            number of classes.
+        label (torch.Tensor): The learning label with shape (N, C), where C is
+            the number of classes.
+        weight (torch.Tensor, optional): Weight of loss. Defaults to None.
+        reduction (str, optional): The method used to reduce the loss.
+            Options are "none", "mean" and "sum". Defaults to 'mean'.
+        avg_factor (int, optional): Average factor that is used to average
+            the loss. Defaults to None.
+
+    Returns:
+        torch.Tensor: The calculated loss
+    """
+    assert pred.dim() == label.dim()
+
+    loss = F.binary_cross_entropy_with_logits(pred, label, reduction='none')
+    loss = loss.sum(dim=1)
+
+    # apply weights and do the reduction
+    if weight is not None:
+        weight = weight.float()
+    loss = weight_reduce_loss(
+        loss, weight=weight, reduction=reduction, avg_factor=avg_factor)
+    return loss
+
+
 @LOSSES.register_module()
 class CrossEntropyLoss(nn.Module):
 
-    def __init__(self, reduction='mean', loss_weight=1.0):
+    def __init__(self, use_sigmoid=False, reduction='mean', loss_weight=1.0):
+        """Cross entropy loss
+
+        Args:
+            use_sigmoid (bool, optional): Whether the prediction uses sigmoid
+                of softmax. Defaults to False.
+            reduction (str, optional): The method used to reduce the loss.
+            Options are "none", "mean" and "sum". Defaults to 'mean'.
+            loss_weight (float, optional):  Weight of the loss.
+                Defaults to 1.0.
+        """
         super(CrossEntropyLoss, self).__init__()
+        self.use_sigmoid = use_sigmoid
         self.reduction = reduction
         self.loss_weight = loss_weight
 
-        self.cls_criterion = cross_entropy
+        if self.use_sigmoid:
+            self.cls_criterion = binary_cross_entropy
+        else:
+            self.cls_criterion = cross_entropy
```

Comment:
BCE is for binary classification. Cross entropy is for multi-classification.
Before BCE, prediction must be normalized by sigmoid function, so BCE_with_sigmoid just combines sigmoid and BCE. However, for multi-classification, before cross entropy, prediction must be normalized by softmax function.